core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts

```
@@ -295,6 +295,10 @@ export class SkillEditorStateService {
   setSkillRights(skillRights: SkillRights): void {
     this._setSkillRights(skillRights);
   }
+
+  getPrerequisiteSkillIds(): string[] {
+    return this._skill ? this._skill.getPrerequisiteSkillIds() : [];
```

Why add this ternary operator? Moreover, why add this function in the first place?

---

> Why add this ternary operator? Moreover, why add this function in the first place?

When I was working on the code, I encountered several instances where the `_skill` object turned out to be null or undefined. This was causing issues and thus, to tackle this, I thought it would be a good idea to add a check using a ternary operator in the `getPrerequisiteSkillIds()` function. So, if `_skill` is null or undefined, instead of causing an error, the function now safely returns an empty array. So in a way this small tweak ensures that the function behaves predictably in all situations, handling potential null reference problems without requiring extensive changes to the rest of the code.

---

@Jaimin791 This isn't the way to handle this, I'm afraid. You'll need to figure out why this._skill is null or undefined, rather than hiding the error, and fix or explain/document the underlying bug. Otherwise this will end up being hacks built on top of hacks and become hard to maintain.